### PR TITLE
Removed use of upward uv flux and added new unit tests.

### PR DESCRIPTION
**Pull request by Kat-90**
Addresses #GitHubissuenum

Description:  
The upward UV flux has been removed from the UV_index calculation along with associated unit tests. Additional unit tests have been added to account for possible negative or NaN values within the input data.
Acceptance test data has been updated.
Related issue is linked to this PR: #1529 

Testing:
 - [X] Ran tests and they passed OK
 - [X] Added new tests for the new feature(s)

CLA
 - [ ] If a new developer, signed up to CLA

### CODE REVIEW

File: improver/uv_index.py [hunk 1]
```diff
@@ -76,9 +69,9 @@ def calculate_uv_index(
...
 
     Raises:
-        ValueError: If uv_upward is not named correctly.
         ValueError: If uv_downward is not named correctly.
-        ValueError: If units do not match.
+        ValueError: If uv_downward contains values that are negative or
+        not a number.
 
     References:
         Turner, E.C, Manners, J. Morcette, C. J, O'Hagan, J. B,
```
Comment (MoseleyS) [inline improver/uv_index.py:77]: Not your fault, but please correct this spelling:
```suggestion
        Turner, E.C, Manners, J. Morcrette, C. J, O'Hagan, J. B,
```

**Review by MoseleyS — CHANGES_REQUESTED** [review]
Hi Kat. 
Your changes look fine, but I've spotted a pre-existing spelling mistake.
Also, I'm surprised that there is no check that the units are in "W m-2" as the docstring requires. It isn't in the CLI either. The default scale factor requires the data to be in these units, so we ought to have a `uv_downward.convert_units("W m-2")` statement.
Also, the units of scale factor are not "no units", but the inverse of the UV data, so "m2 W-1".
Is it unreasonable to make changes to do this too? I know it is out of the original scope of the ticket.

File: improver/uv_index.py [hunk 2]
```diff
@@ -97,12 +97,13 @@ def calculate_uv_index(
...
         raise ValueError(msg)
 
+    uv_downward.convert_units("W m-2")
     uv_data = uv_downward.data * scale_factor
     attributes = generate_mandatory_attributes(
         [uv_downward], model_id_attr=model_id_attr
     )
     uv_index = create_new_diagnostic_cube(
-        "ultraviolet_index", "1", uv_downward, attributes, data=uv_data
+        "ultraviolet_index", "W m-2", uv_downward, attributes, data=uv_data
```
Comment (MoseleyS) [inline improver/uv_index.py:106]: No. The units for the UV index were correctly unitless. The point of the scale factor is that a UV index of 10 relates to a UV flux of 10/3.6 W m-2.

**Review by MoseleyS — CHANGES_REQUESTED** [review]

**Review by MoseleyS — DISMISSED** [review]

File: improver_tests/uv_index/test_uv_index.py [hunk 3]
```diff
@@ -70,46 +61,71 @@ def test_basic(self):
...
-        """Tests that a ValueError is raised if the input uv files have
-        different units. """
-        msg = "The input uv files do not have the same units."
-        with self.assertRaisesRegex(ValueError, msg):
-            calculate_uv_index(self.cube_uv_up, self.cube_diff_units, scale_factor=1.0)
-
     def test_badname_down(self):
         """Tests that a ValueError is raised if the input uv down
         file has the wrong name. """
         msg = "The radiation flux in UV downward"
```
Comment (bayliffe) [inline improver_tests/uv_index/test_uv_index.py:87]: This test actually needs a slightly longer expected, as there are two exceptions that both raise an error starting with "The radiation flux in UV downward", meaning this test cannot currently differentiate between them and check the correct exception has been triggered.

File: improver_tests/uv_index/test_uv_index.py [hunk 4]
```diff
@@ -70,46 +61,71 @@ def test_basic(self):
...
+        negative_data_down = np.full_like(
+            self.cube_uv_down.data, dtype=np.float32, fill_value=-0.1
+        )
+        uv_down_name = "surface_downwelling_ultraviolet_flux_in_air"
+        negative_uv_down = set_up_variable_cube(
+            negative_data_down, name=uv_down_name, units="W m-2"
+        )
+        msg = (
+            "The radiation flux in UV downward contains data "
+            "that is negative or NaN. Data should be > 0."
```
Comment (bayliffe) [inline improver_tests/uv_index/test_uv_index.py:103]:
```suggestion
            "that is negative or NaN. Data should be >= 0."
```

File: improver_tests/uv_index/test_uv_index.py [hunk 5]
```diff
@@ -70,46 +61,71 @@ def test_basic(self):
...
+    def test_negative_input(self):
+        """Tests that a ValueError is raised if the input contains
+        negative values. """
+        negative_data_down = np.full_like(
+            self.cube_uv_down.data, dtype=np.float32, fill_value=-0.1
+        )
+        uv_down_name = "surface_downwelling_ultraviolet_flux_in_air"
+        negative_uv_down = set_up_variable_cube(
+            negative_data_down, name=uv_down_name, units="W m-2"
+        )
```
Comment (bayliffe) [inline improver_tests/uv_index/test_uv_index.py:100]: Or just:
```suggestion
        negative_uv_down = self.cube_uv_down.copy(data=negative_data_down)
```

File: improver_tests/uv_index/test_uv_index.py [hunk 6]
```diff
@@ -70,46 +61,71 @@ def test_basic(self):
...
+        with self.assertRaisesRegex(ValueError, msg):
+            calculate_uv_index(negative_uv_down)
+
+    def test_nan_input(self):
+        """Tests that a ValueError is raised if the input contains
+        values that are not a number. """
+        self.cube_uv_down.data.fill(np.nan)
+        msg = (
+            "The radiation flux in UV downward contains data "
+            "that is negative or NaN. Data should be > 0."
```
Comment (bayliffe) [inline improver_tests/uv_index/test_uv_index.py:114]:
```suggestion
            "that is negative or NaN. Data should be >= 0."
```

**Review by bayliffe — CHANGES_REQUESTED** [review]
Looks good, a few tiny tweaks and suggestions.

File: improver/uv_index.py [hunk 7]
```diff
@@ -67,7 +60,7 @@ def calculate_uv_index(
             been empirically derived and should not be
             changed except if there are scientific reasons to
             do so. For more information see section 2.1.1 of the paper
-            referenced below (no units)
+            referenced below.(m2 W-1)
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/uv_index.py]: Not quite what I meant, I was  more thinking:
```
        scale_factor:
            The uv scale factor. Default is 3.6 (m2 W-1). This factor has
            been empirically derived and should not be
            changed except if there are scientific reasons to
            do so. For more information see section 2.1.1 of the paper
            referenced below.
```

[the remark this answers is not in the document]
Comment (Kat-90) [reply improver/uv_index.py]: Sorry, I see what you meant now. Done.

**Review by bayliffe — APPROVED** [review]

**Pull request merged**